sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuExec.scala

```diff
@@ -108,20 +108,14 @@ object GpuMetric extends Logging {
     case i => throw new IllegalArgumentException(s"found unsupported GpuMetric ${i.getClass}")
   }
 
-  def unwrap(input: Map[String, GpuMetric]): Map[String, SQLMetric] = input.filter {
-    // remove the metrics that are not registered
-    case (_, NoopMetric) => false
-    case _ => true
-    // sadly mapValues produces a non-serializable result, so we have to hack it a bit to force
-    // it to be materialized
-  }.mapValues(unwrap).toArray.toMap
-
-  def wrap(input: SQLMetric): GpuMetric = WrappedGpuMetric(input)
+  def unwrap(input: Map[String, GpuMetric]): Map[String, SQLMetric] = input.collect {
+    // remove the metrics that are not registered (NoopMetric)
+    case (k, w: WrappedGpuMetric) => (k, w.sqlMetric)
```

Comment:
This code makes the assumption that there's only two implementations of GpuMetric, Wrapped and Noop.  There's was already a method that can unwrap any GpuMetric, so it should be leveraged here to make this a bit more future-proof.  Doing so more closely aligns the code to the comment (and arguably the comment is no longer necessary since it's more obvious what it is doing after updating the code).  Most importantly, updating the code means we proactively fail if someone adds a new GpuMetric derivation but forgets to update the unwrap method.  Otherwise the code silently strips the new metric type out of the map when unwrapping which could likely go unnoticed.
```suggestion
    // remove the metrics that are not registered (NoopMetric)
    case (k, w) if !w.isInstanceOf[NoopMetric] => (k, unwrap(w))
```